### ISSUE TITLE
Autocomplete doesn't work inside a function

### ISSUE THREAD

**Issue by josephtocci** [workflowproducts/postage#82]
Autocomplete doesn't show up when I need it:
````

    str_ret := (SELECT rinv_header.??? FROM usr.rinv_header WHERE id = int_id);
END
$BODY$
````

**Comment by michaeltocci** [comment]
Fixed, will be seen in the next release.

**TL;DR**
The issue was that the query detection code I wrote was unable to pick up the query you wanted to autocomplete for. I re-wrote the query detection to start from the cursor instead of starting at the top of the script. I also removed some `PL/pgSQL` specific code, which increases reliability.

**The long story**
The first version of the query detection code I wrote was very complex and easily confused. It worked by doing "scans" of the code (starting at the top of the script), and the code was thought about as if they were levels (a `SELECT` inside an `IF` inside a `CREATE FUNCTION` would be a third level statement if I'm not mistaken). This approach was slow and had a lot of code that dealt with specifics of `PL/pgSQL` (and being language specific meant that you would have trouble with SQL autocomplete within, say, a `PL/Python` function).

To solve this I re-wrote the query detection code. Now, to detect the query I go backwards from the cursor attempting to find a `Postgres` (not `PL/pgSQL`) keyword that starts a query (`DROP`, `CREATE`, `SELECT`, `INSERT`, `GRANT`, `REVOKE` Etc...). Once I find that keyword a couple of things happen, first if the keyword I found was any of these: `EXECUTE`, `SELECT`, `INSERT`, `UPDATE`, `DELETE` or `TRUNCATE` I do some extra processing to verify that we are not in a `GRANT` or a `REVOKE` query (if we are then I use that `GRANT` or `REVOKE` as the starting keyword). Next, to get the end of the query, (starting at the keyword) I use a variant of our SQL split code to find the next uncommented semi-colon or the EOF.

Once I have this range, I check to see if the cursor is within the query
If it is, we're found the correct query and we're done.
If it isn't, I run the same keyword search starting from behind the query we just found. However, instead of searching for any query-starting keyword, I search only for the starting keywords of queries that have a possibility of encompassing other queries (`COMMENT`, `COPY`, `CREATE`, `DELETE`, `DO`, `EXPLAIN`, `INSERT`, `NOTIFY`, `PREPARE`, `SELECT`, `UPDATE` and `VALUES`), this increases speed.

Now, there are two cases where this code is sluggish (even though it's still faster than it's predecessor):

- If your selecting a query that has lots of sub-queries between it's start and where the cursor is
- If you're cursor is low in a file with lots of queries and your cursor is not within any queries
